### ISSUE TITLE
flshattF@v2.0.8` is not supported because: attn_bias type is <class 'torch.Tensor'>

### ISSUE THREAD

**Issue by mashrurmorshed** [facebookresearch/xformers#855]
# ❓ Questions and Help

I couldn't use attention bias with memory efficient attention. The error I get is:

```
Operator `memory_efficient_attention` does not support inputs:
     query       : shape=(128, 4096, 1, 16) (torch.float16)
     key         : shape=(128, 197, 1, 16) (torch.float16)
     value       : shape=(128, 197, 1, 16) (torch.float16)
     attn_bias   : <class 'torch.Tensor'>
     p           : 0.0
`flshattF@v2.0.8` is not supported because:
    attn_bias type is <class 'torch.Tensor'>
```
The shape of my attention bias is: `([128, 1, 4096, 197])`.

Checking the stack trace, I get to [xformers/ops/fmha/common.py#L242](https://github.com/facebookresearch/xformers/blob/4cf516290ff341a280954ea41c66e13bece50228/xformers/ops/fmha/common.py#L242):

```python
if type(d.attn_bias) not in cls.SUPPORTED_ATTN_BIAS_TYPES
```

It seems the supported bias types are:

```python
SUPPORTED_ATTN_BIAS_TYPES: Set[Any] = {
      type(None),
      LowerTriangularMask,
      BlockDiagonalMask,
      BlockDiagonalCausalMask,
      BlockDiagonalCausalFromBottomRightMask,
  }
```

Is it not possible to use attention bias with a tensor mask?

**Comment by mashrurmorshed** [comment]
I could use it ultimately with the [Cutlass backend](https://github.com/facebookresearch/xformers/blob/4cf516290ff341a280954ea41c66e13bece50228/xformers/ops/fmha/cutlass.py#L158), which seems to support `torch.tensor` bias.

**Comment by danthe3rd** [comment]
Yes that's correct. Also more generally we recommend that you don't enforce the backend you want to use, so xFormers can select the best performing one for your problem/hardware (eg don't specify "op")

**Comment by mashrurmorshed** [comment]
Thanks, that makes sense. I was using it through another library that was specifying the op as `xformers.ops.MemoryEfficientAttentionFlashAttentionOp`, which was causing the issue. I'll go ahead and close this.
